File: 1236-NThTribonacciNumber/1236-NThTribonacciNumber.java
// Last updated: 2026. 7. 16. 오전 11:25:56
class Solution {
    public int tribonacci(int n) {
        int[] t = new int[38];
        t[0] = 0;
        t[1] = 1;
        t[2] = 1;
        for (int i=3; i<=37; ++i) {
            t[i] = t[i-3] + t[i-2] + t[i-1];
        }
        return t[n];
    }
}

File: 1236-NThTribonacciNumber/1236-NThTribonacciNumber.ts
// Last updated: 2026. 7. 16. 오전 11:25:51
function tribonacci(n: number): number {
    let one: number = 0;
    let two: number = 1;
    let three: number = 1;
    if (n == 0) {
        return one;
    }
    if (n == 1) {
        return two;
    }
    n -= 2;
    while (n-- > 0) {
        let temp: number = one + two + three;
        one = two;
        two = three;
        three = temp;
    }
    return three;
};